projects/fw_102_hw/src/main.c

```diff
@@ -0,0 +1,41 @@
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+
+  storage->counter_a++;
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a);
+
+  if (storage->counter_a % 2 == 0) {
+    storage->counter_b++;
+    LOG_DEBUG("Counter B: %i\n", storage->counter_b);
+  }
+
+  // start another timer
+  soft_timer_start_millis(500, prv_timer_callback, storage, NULL);
```

Comment:
Please move the constant 500 here and below into a `#define`'d macro symbol with a descriptive name including the unit. That makes it easier to tell what the 500 is as opposed to a magic number, and makes it easier to change by consolidating a constant that currently appears in 2 places into 1.